networking/hub-regionA.json

```diff
@@ -98,89 +316,132 @@
                     {
                         "name": "GatewaySubnet",
                         "properties": {
-                            "addressPrefix": "[parameters('azureGatewaySubnetAddressSpace')]",
-                            "serviceEndpoints": [
-                            ]
+                            "addressPrefix": "[parameters('azureGatewaySubnetAddressSpace')]"
                         }
                     },
                     {
                         "name": "AzureBastionSubnet",
                         "properties": {
                             "addressPrefix": "[parameters('azureBastionSubnetAddressSpace')]",
-                            "serviceEndpoints": [
-                            ]
+                            "networkSecurityGroup": {
+                                "id": "[resourceId('Microsoft.Network/networkSecurityGroups', variables('bastionNetworkNsgName'))]"
+                            }
                         }
                     }
-                ],
-                "enableDdosProtection": false,
-                "enableVmProtection": false
-            }
+                ]
+            },
+            "resources": [
+                {
+                    "type": "providers/diagnosticSettings",
+                    "apiVersion": "2017-05-01-preview",
+                    "name": "Microsoft.Insights/default",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.OperationalInsights/workspaces', variables('hubLaName'))]",
+                        "[resourceId('Microsoft.Network/virtualNetworks', variables('hubVnetName'))]"
+                    ],
+                    "properties": {
+                        "workspaceId": "[resourceId('Microsoft.OperationalInsights/workspaces', variables('hubLaName'))]",
+                        "metrics": [
+                            {
+                                "category": "AllMetrics",
+                                "enabled": true
+                            }
+                        ]
+                    }
+                }
+            ]
         },
         {
-            "apiVersion": "2019-11-01",
             "type": "Microsoft.Network/publicIpAddresses",
-            "name": "[variables('defaultFwPipName')]",
+            "apiVersion": "2020-05-01",
+            "name": "[variables('hubFwPipNames')[copyIndex()]]",
             "location": "[parameters('location')]",
+            "comments": "This is a public IP for this regional hub's firewall. You'll want as many as necessary to avoid SNAT port exhaustion. Typical production hubs may require 20 IPs -- we only deploy three here as a reference.",
```

Comment:
Same comment here as for hub-default, a link on SNAT exhaustion would be good.

Reply:
Same response :)